### ISSUE TITLE
Regression in Unit_hipFreeNegativeArray

### ISSUE THREAD

**Issue by Kerilk** [CHIP-SPV/chipStar#547]
https://github.com/CHIP-SPV/chipStar/commit/34f5f903db7ca78f41f791f64a07a8684c326162 introduced a regression in `Unit_hipFreeNegativeArray`

```
665/953 Test #665: Unit_hipFreeNegativeArray .................................................***Failed    0.54 sec
Filters: Unit_hipFreeNegativeArray

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hipFree is a Catch v2.13.4 host application.
Run with -? for options

-------------------------------------------------------------------------------
Unit_hipFreeNegativeArray
  ArrayFree
-------------------------------------------------------------------------------
/gpfs/jlse-fs0/users/bertoni/chip-spv-novolatile/chip-spv/HIP/tests/catch/unit/memory/hipFree.cc:181
...............................................................................

/gpfs/jlse-fs0/users/bertoni/chip-spv-novolatile/chip-spv/HIP/tests/catch/unit/memory/hipFree.cc:181: FAILED:
  REQUIRE( localError == hipErrorInvalidValue )
with expansion:
  709 == 1
with message:
  Matching Errors:
      Expected Error: hipErrorInvalidValue
      Expected Code: 1
                    Actual Error:   hipErrorUnknown
      Actual Code:   709
  Str: hipFreeArray(nullptr)
      In File: /gpfs/jlse-fs0/users/bertoni/chip-spv-novolatile/chip-spv/HIP/
  tests/catch/unit/memory/hipFree.cc
      At line: 181
```

```
SECTION(“ArrayFree”) { HIP_CHECK_ERROR(hipFreeArray(nullptr), hipErrorInvalidValue); }
```

**Comment by pvelesko** [comment]
This will be fixed in my #535  branch which I hope to cleanup before the release EOD today

**Comment by pvelesko** [comment]
How did this come up? Are you running check.py? PRs get tested on CI

**Comment by Kerilk** [comment]
I only see the linter being run for CI for the last week or so

**Comment by pvelesko** [comment]
CI gets run before every merge to main

**Comment by Kerilk** [comment]
was merged in RC1

**Comment by Kerilk** [comment]
Not sure last minute regression should be addressed in 1.1
